lib/Optimizer/Transforms/QuakeAddMetadata.cpp

```diff
@@ -120,20 +120,45 @@ struct QuakeFunctionAnalysis {
       // Handle auto reg = mz(q); reset(q)
       // don't necessarily need conditional statements
       funcOp->walk([&](Operation *op) {
-        if (!isa<quake::MzOp>(op))
+        if (!isa<quake::MxOp, quake::MyOp, quake::MzOp>(op))
           return WalkResult::skip();
 
-        auto mzOp = dyn_cast<quake::MzOp>(op);
-        if (mzOp.getRegisterName().has_value()) {
-          for (auto measuredQubit : mzOp.getTargets()) {
-            for (auto user : measuredQubit.getUsers()) {
-              if (dyn_cast<quake::ResetOp>(user)) {
-                data.hasConditionalsOnMeasure = true;
-                return WalkResult::interrupt();
+        if (auto mxOp = dyn_cast<quake::MxOp>(op))
+          if (mxOp.getRegisterName().has_value()) {
+            for (auto measuredQubit : mxOp.getTargets()) {
+              for (auto user : measuredQubit.getUsers()) {
+                if (dyn_cast<quake::ResetOp>(user)) {
+                  data.hasConditionalsOnMeasure = true;
+                  return WalkResult::interrupt();
+                }
               }
             }
           }
-        }
+
+        if (auto myOp = dyn_cast<quake::MyOp>(op))
+          if (myOp.getRegisterName().has_value()) {
+            for (auto measuredQubit : myOp.getTargets()) {
+              for (auto user : measuredQubit.getUsers()) {
+                if (dyn_cast<quake::ResetOp>(user)) {
+                  data.hasConditionalsOnMeasure = true;
+                  return WalkResult::interrupt();
+                }
+              }
+            }
+          }
+
+        if (auto mzOp = dyn_cast<quake::MzOp>(op))
+          if (mzOp.getRegisterName().has_value()) {
+            for (auto measuredQubit : mzOp.getTargets()) {
+              for (auto user : measuredQubit.getUsers()) {
+                if (dyn_cast<quake::ResetOp>(user)) {
+                  data.hasConditionalsOnMeasure = true;
+                  return WalkResult::interrupt();
+                }
+              }
+            }
+          }
```

Comment:
This could be turned into a lambda (or template) instead of block copying the code.